src/main/java/io/vertx/httpproxy/HttpProxy.java

```diff
@@ -39,8 +40,8 @@ public interface HttpProxy extends Handler<HttpServerRequest> {
    * @param client the {@code HttpClient} that forwards <i><b>outbound</b></i> requests to the <i><b>origin</b></i>.
    * @return a reference to this, so the API can be used fluently.
    */
-  static HttpProxy reverseProxy(HttpClient client) {
-    return new ReverseProxy(new ProxyOptions(), client);
+  static HttpProxy reverseProxy(HttpClient client, Vertx vertx) {
```

Comment:
It might be useful to move the `HttpClientInternal` interface to the `io.vertx.core.internal.http` in Vert.x core.

Then we would be able to cast the client to `HttpClientInternal` and invoke `HttpClientInternal.vertx()`. That would avoid this breaking change.

Reply:
should I create a PR for vert.x core first?